### ISSUE TITLE
The new splitter crashes with no arguments

### ISSUE THREAD

**Issue by permcody** [libMesh/libmesh#1190]
It looks like there was an attempt to try to provide help but it wasn't a very good attempt 😄.

If you run the new "splitter" without any arguments, instead of printing the help message that's in the source, it just crashes:

```
#0  0x00007fffe6c95e75 in waitpid () from /lib64/libc.so.6
#1  0x00007fffe6c29cd1 in do_system () from /lib64/libc.so.6
#2  0x00007fffeca2d3b6 in libMesh::print_trace(std::ostream&) () from /home/permcj/projects/falcon/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#3  0x00007fffeca29d5a in libMesh::MacroFunctions::report_error(char const*, int, char const*, char const*) () from /home/permcj/projects/falcon/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#4  0x00007fffece72704 in libMesh::NameBasedIO::read(std::string const&) () from /home/permcj/projects/falcon/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#5  0x00007fffeceb2c28 in libMesh::UnstructuredMesh::read(std::string const&, libMesh::MeshData*, bool, bool) () from /home/permcj/projects/falcon/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#6  0x00000000004089cd in main ()
```

@friedmud

**Comment by roystgnr** [comment]
I get

```
$ ./splitter-dbg 
Reading 
 ERROR: Unrecognized file extension: 
   I understand the following:

     *.e    -- Sandia's ExodusII format
     *.exd  -- Sandia's ExodusII format
     *.gmv  -- LANL's General Mesh Viewer format
     *.mat  -- Matlab triangular ASCII file
     *.n    -- Sandia's Nemesis format
     *.nem  -- Sandia's Nemesis format
     *.off  -- OOGL OFF surface format
     *.ucd  -- AVS's ASCII UCD format
     *.unv  -- I-deas Universal format
     *.vtu  -- Paraview VTK format
     *.inp  -- Abaqus .inp format
     *.xda  -- libMesh ASCII format
     *.xdr  -- libMesh binary format
     *.gz   -- any above format gzipped
     *.bz2  -- any above format bzip2'ed
     *.xz   -- any above format xzipped
     *.cpa  -- libMesh Checkpoint ASCII format
     *.cpr  -- libMesh Checkpoint binary format
```

followed by roughly the same stack trace and then the post-exit printout.  Looks correct to me.

**Comment by permcody** [comment]
If you look at the source, there's actually a help message in there but it's only printed if "--help" is passed. This utility requires arguments so it should print the help message if none are passed.

**Comment by jwpeterson** [comment]
Fixed by 00ec25c007a1b9c9b763ba8c2a4c4a1b32c56daf.
